pkg/sql/parser/format.go

```diff
@@ -112,13 +120,27 @@ type NodeFormatter interface {
 	Format(buf *bytes.Buffer, flags FmtFlags)
 }
 
+// formatOrScrubNode recurses into a node for pretty-printing, unless
+// scrubDatums is set in the flags and the node is a datum or a
+// literal.
+func formatOrScrubNode(buf *bytes.Buffer, f FmtFlags, n NodeFormatter) {
+	if f.scrubDatums {
+		switch n.(type) {
+		case Datum, *NumVal, *StrVal:
```

Comment:
Can we match `Constant` here? if not, we should add a comment there to update this if adding another one.